File: src/main/java/com/skidders/sigma/utils/misc/StringUtil.java
package com.skidders.sigma.utils.misc;

import net.minecraft.client.Keyboard;
import net.minecraft.client.MinecraftClient;
import net.minecraft.client.options.KeyBinding;
import net.minecraft.client.util.InputUtil;

public class StringUtil {

    public static String convertKeyToName(int keyCode) {
        for (Keys key : Keys.values()) {
            if (key.keyCode == keyCode) {
                return key.name;
            }
        }

        InputUtil.Key input = InputUtil.fromKeyCode(keyCode, 0);
        String[] translationKeyParts = input.getTranslationKey().split("\\.");

        if (translationKeyParts.length > 0) {
            String keyName = translationKeyParts[translationKeyParts.length - 1];

            if (!keyName.isEmpty()) {
                String prefix = keyCode <= 4 ? "Mouse " : "";
                return prefix + keyName.substring(0, 1).toUpperCase() + keyName.substring(1);
            }
        }

        return "Unknown";
    }

}

File: src/main/kotlin/com/skidders/sigma/utils/misc/StringUtil.kt
package com.skidders.sigma.utils.misc;

import net.minecraft.client.util.InputUtil;
import java.util.Locale

object StringUtil {
    fun convertKeyToName(keyCode: Int): String {
        for (key in Keys.entries) {
            if (key.keyCode == keyCode) {
                return key.key;
            }
        }

        val input = InputUtil.fromKeyCode(keyCode, 0);
        val translationKeyParts = input.translationKey.split("\\.");

        if (translationKeyParts.count() > 0) {
            val keyName = translationKeyParts[translationKeyParts.count() - 1];

            if (!keyName.isEmpty()) {
                val prefix = if (keyCode <= 4) "Mouse " else "";
                return prefix + keyName.substring(0, 1).uppercase(Locale.getDefault()) + keyName.substring(1);
            }
        }

        return "Unknown";
    }
}
